src/sync/pull.rs

```diff
@@ -301,9 +319,16 @@ pub async fn maybe_end_try_pull(
             main_snapshot.value_hash()
         );
     }
+    let main_diff = diff.map_err(InvalidUtf8)?;
+    let mut diffs = BTreeMap::new();
```

Comment:
What about the index diffs? We need those too.

Seems like the code for computing the set of index diffs should be pulled out so it can be used by both CommitTransaction and Pull.

Reply:
oops

Reply:
We do not seem to update the index at all in this code path? Maybe it is done somewhere else?

Reply:
Figured out what is going on. We update the index when applying the patch and then again when we replay the mutations. I will have to load the indexes from before the patch and the indexes from after all the replays and diff all of them.

Reply:
Yeah the indexes must be recomputed to match the state before each replay begins (when the mutation runs, it must see the indexes in the correct state to correspond to that cache state).

> I will have to load the indexes from before the patch and the indexes from after all the replays and diff all of them.

Yep!